compiler/test/suites/pattern_matching.re

```diff
@@ -357,6 +357,34 @@ describe("pattern matching", ({test, testSkip}) => {
     |},
     "7\n",
   );
+  assertRun(
+    "destructure_mut_tuple",
+    {|
+      let run = () => {
+        let mut (a, b) = (1, 2)
+
+        print(b)
+      }
+
+      run()
+    |},
+    "2\n",
+  );
+  assertRun(
+    "destructure_mut_tuple_closure",
+    {|
+      let run = () => {
+        let mut (a, b) = (1, 2)
+
+        print(b)
+
+        () => b
+      }
+
+      run()
+    |},
+    "2\n",
+  );
```

Comment:
These tests have me wondering about this:
```
let run = () => {
        let mut a = 1
        let mut b = 2
        let mut (x, y) = (a, b)
        b = 7
        print(y)
}
run()
```
What happens?

Reply:
You get `2`!

Reply:
Can you add that a test?

Reply:
Done!